pkgs/os-specific/linux/realmd/default.nix

```diff
@@ -0,0 +1,66 @@
+{ stdenv
+, fetchFromGitLab
+, openldap
+, libkrb5
+, packagekit
+, polkit
+, libxslt
+, intltool
+, glib
+, pkg-config
+, systemd
+, autoreconfHook
+, samba
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "realmd";
+  version = "0.17.0";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "1c6q2a86kk2f1akzc36nh52hfwsmmc0mbp6ayyjxj4zsyk9zx5bf";
+  };
+
+  # I didn't know substitueInPlace was a thing...
+  postUnpack = ''
+    gawk -i inplace
+         'BEGIN {FS="=";}
+         /paths/ {
+           flag=1;print;next
+         };
+         /active-directory|^\s$/ {
+           flag=0; next
+         };
+         flag == 1 && $1 {
+            if ($1 ~ "net") {
+               print $1, "=", "${samba}/sbin/net";
+               next;
+            }
+            if ($1 ~ "winbindd") {
+               print $1, "=", "${samba}/sbin/winbindd";
+               next;
+            }
+            # if ($1 ~ "adcli") {
+            #    print $1, "=", "$adcli}/sbin/adcli";
+            #    next;
+            # }
+         };
+         {print};' $sourceRoot/service/realmd-defaults.conf
```

Comment:
Boi, such a huge script would be more useful if you did it a separate file and included it.